features/support/commits_finder.rb

```diff
@@ -0,0 +1,49 @@
+# Encapsulates finding Git commits in flexible ways
+#
+# After initializing with the desired attributes for the commit list,
+# run as many #find_* methods as you want to build up your commit list.
+# When done, call #to_a to get the results as a Cucumber compatible array.
+class CommitsFinder
```

Comment:
I get better what you say now. I have refactored this the following way:
* all CommitFinder methods now always return a CommitsList object
* the CommitsList object allows queries whether it is empty, or to provide it as a Cucumber table

I agree that this is a much better setup. Thanks for holding the ground on this!  :)